konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/classifier/ClassifierOutputRunner.java

```diff
@@ -0,0 +1,201 @@
+
+        String inputName = step.inputName();
+
+
+        if (inputName == null) {
+            String errMultipleKeys = "NDArray field name was not provided and could not be inferred: multiple NDArray fields exist: %s and %s";
+            String errNoKeys = "NDArray field name was not provided and could not be inferred: no image NDArray exist";
+            inputName = DataUtils.inferField(data, ValueType.NDARRAY, false, errMultipleKeys, errNoKeys);
+        }
+
+
+        NDArray classifierOutput = data.getNDArray(inputName);
+
+        boolean batch = false;
+        if (classifierOutput.shape().length == 2 && classifierOutput.shape()[0] > 1) {
+            batch = true;
+        }
+
+        // If not specified, the predicted class index as a string is used - i.e., "0", "1"
+        List<String> labels = step.Labels();
+
+        if (labels == null) {
+            labels = new ArrayList<String>();
+        }
+        if (labels.isEmpty()) {
+            for (int i = 0; i < classifierOutput.shape()[1]; i++) {
+                labels.add(Integer.toString(i));
+            }
+        }
+
+
+        if (!batch) {
+            double[] classifierOutputArr = squeze(classifierOutput);
+            double[] maxValueWithIdx = getMaxValueAndIndex(classifierOutputArr);
+            double prob = maxValueWithIdx[0];
+            long index = (long) maxValueWithIdx[1];
+            String label = labels.get((int) index);
+
+            if (step.topN() != null && step.topN() > 1) {
+                if (step.returnProb()) {
+                    data.putListDouble(step.probName(), Collections.singletonList(prob));
+                }
+                if (step.returnIndex()) {
+                    data.putListInt64(step.indexName(), Collections.singletonList(index));
+                }
+                if (step.returnLabel()) {
+                    data.putListString(step.labelName(), Collections.singletonList(label));
+                }
+            } else {
+
+                if (step.returnProb()) {
+                    data.put(step.probName(), prob);
+                }
+                if (step.returnIndex()) {
+                    data.put(step.indexName(), index);
+                }
+                if (step.returnLabel()) {
+                    data.put(step.labelName(), label);
+                }
+
+            }
+            if (step.allProbabilities()) {
+                data.putListDouble("allProbabilities", DoubleStream.of(classifierOutputArr).boxed().collect(Collectors.toList()));
+            }
+        }
+
+        if (batch) {
+
+            int bS = (int) classifierOutput.shape()[1];
+            double[][] y = classifierOutput.getAs(double[][].class);
+
+            List<Double> probs = new ArrayList<Double>();
+            List<Long> indeces = new ArrayList<Long>();
+            List<String> labelsList = new ArrayList<String>();
+            List<NDArray> allPropabilities = new ArrayList<NDArray>();
```

Comment:
We'll use `List<List<Double>>` for consistency with List<Double> for the non-batch case
so `double[][] arr = ...` then loop over each `arr[i]` (each row)